core/src/test/java/de/evoila/cf/broker/controller/core/CatalogControllerTest.java

```diff
@@ -0,0 +1,136 @@
+        }
+        return generatedServices;
+    }
+
+    /**
+     * A function testing CatalogController.getCatalog with a valid catalog.
+     * The catalog is stored in a JSON file.
+     * Not an @Test but called in @Test methods.
+     * @param catalogPath           The path of the JSON file containing the catalog.
+     * @param requestIdentity       A string representing a request identity.
+     * @param originatingIdentity   A string representing a originating identity.
+     * @throws Exception
+     */
+    private void validCatalogTest(String catalogPath,
+                                  String requestIdentity,
+                                  String originatingIdentity) throws Exception
+    {
+        // mocks
+        when(catalogService.getCatalog()).thenReturn(catalog);
+
+        List<ServiceDefinition> generatedServices = readServicesFile(catalogPath);
+        when(catalog.getServices()).thenReturn(generatedServices);
+        // actual method call
+        ResponseEntity<Catalog> response = catalogController.getCatalog(requestIdentity,
+                originatingIdentity);
+        // create expected object
+        Catalog expectedCatalog = new Catalog();
+        List<ServiceDefinition> expectedServices = readServicesFile(catalogPath);
+        expectedCatalog.setServices(expectedServices);
+        // check results
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertEquals(expectedCatalog, response.getBody()); // ist dieser Test wirklich sinnvoll? Wir mocken ja gerade getServices. Wäre es ausreichend zu prüfen, dass der Catalog nicht null ist? Oder nicht leer? In diesen Fällen bräuchte man auch die ganzen equals Funktionen nicht hinzufügen.
+    }
+
+    @Test
+    void getCatalog_EmptyCatalog() {
+        // mocks
+        when(catalogService.getCatalog()).thenReturn(catalog);
+        when(catalog.getServices()).thenReturn(new ArrayList<>());
+        // actual method call
+        ResponseEntity<Catalog> response = catalogController.getCatalog(REQUEST_IDENTITY,
+                                                                        ORIGINATING_IDENTITY);
+        // check results
+        assertEquals(HttpStatus.OK, response.getStatusCode());
+        assertEquals(new Catalog(), response.getBody());
+    }
+
+    @Test
+    void getCatalog_ValidCatalog()
+            throws Exception {
+        validCatalogTest(PATH_VALID_CATALOG_SERVICES,
+                REQUEST_IDENTITY,
+                ORIGINATING_IDENTITY);
+    }
+
+    @Test
+    void getCatalog_ValidCatalogNoIdentityHeaders()
+            throws Exception {
+        validCatalogTest(PATH_VALID_CATALOG_SERVICES,
+                null,
+                null);
+    }
+
+    @Test
+    void getCatalog_ValidCatalogInvalidIdentityFormat()
+            throws Exception {
+        validCatalogTest(PATH_VALID_CATALOG_SERVICES,
+                ORIGINATING_IDENTITY, // expected: GUID, here: platform + base64-String
```

Comment:
Validations to these headers happens in interceptor classes. Therefore does the CatalogController not check these headers on syntax or semantics, but just provides them to the services it calls.

Also this test passes and therefor does fail its purpose, doesn't it?

Reply:
We gonna remove the tests for these cases.